FaceID/faceID.py

```diff
@@ -0,0 +1,361 @@
+                print("Removed Employee ID: ", extractEmpID, " successfully!\n")
+                flag = True
+    
+        if flag==False: 
+            print("Employee ID not found!")
+
+# Trains images in the dataset, outputs an updated .yml file to the trainer/ directory
+def trainDataset():
+
+    photoRegister = readPhotoRegister() # updates global variable from txt file
+    if len(photoRegister)>0: # checks if employees exist
+
+        # grabs face profile tools
+        path = 'dataset'
+        recognizer = cv2.face.LBPHFaceRecognizer_create()
+        detector = cv2.CascadeClassifier("Cascades/haarcascade_frontalface_default.xml")
+
+        # function to get the images and label data
+        def getImagesAndLabels(path):
+            imagePaths = [os.path.join(path,f) for f in os.listdir(path)]
+            faceSamples=[]
+            ids = []
+            for imagePath in imagePaths:
+                PIL_img = Image.open(imagePath).convert('L') # convert it to grayscale
+                img_numpy = np.array(PIL_img,'uint8')
+                id = int(os.path.split(imagePath)[-1].split("_")[0])
+                print(id)
+                faces = detector.detectMultiScale(img_numpy)
+                for (x,y,w,h) in faces: # goes through faces and pairs faces with id
+                    faceSamples.append(img_numpy[y:y+h,x:x+w])
+                    ids.append(id)
+            return faceSamples,ids
+        print ("\n [INFO] Training faces . . .")
+        faces,ids = getImagesAndLabels(path) # gets faces from dataset/ folder
+        recognizer.train(faces, np.array(ids)) # feeds faces to trainer ! 
+        # Save the model into trainer/trainer.yml
+        recognizer.write('trainer/trainer.yml')
+        print("\n Training complete !")
+
+    else: print("\nNo Registered Employees!\n")
+
+
+# Detects and identifies registered employee faces
+def searchFace():
+
+    global PhotoRegister
+    photoRegister = readPhotoRegister() # updates global variable from txt file
+    if len(photoRegister)>0: # checks if employees exist
+
+        trainDataset() # Train face models 
+
+        # loads trained face data and face detection models
+        recognizer = cv2.face.LBPHFaceRecognizer_create()
+        recognizer.read('trainer/trainer.yml')
+        cascadePath = "Cascades/haarcascade_frontalface_default.xml"
+        faceCascade = cv2.CascadeClassifier(cascadePath);
+        font = cv2.FONT_HERSHEY_SIMPLEX
+
+        # Initialize and start realtime video capture
+        cam = cv2.VideoCapture(0)
+        cam.set(3, 640) # set video widht
+        cam.set(4, 480) # set video height
+
+        # Define min window size to be recognized as a face
+        minW = 0.1*cam.get(3)
+        minH = 0.1*cam.get(4)
+
+        print("Press 'ESC' key to exit\n")
+        #EmpID = 0 # stores index for identified face
+        while True:
+            ret, img =cam.read()
+            #img = cv2.flip(img, -1) # Flips cam view vertically 
```

Comment:
Will we want to add this back later? If so, we can use `TODO`s to mark places in our code where we need to do something later (ex: `#TODO: uncomment out flip cam view code once {something} is done)`